File: 954-maximum-sum-circular-subarray/maximum-sum-circular-subarray.cpp
class Solution {
public:
    int maxSubarraySumCircular(vector<int>& nums) {
        int globalMax = nums[0], globalMin = nums[0];
        int currMax = 0, currMin = 0;
        int sum = 0;

        for(int i=0; i<nums.size(); i++){
            currMax = max(currMax + nums[i], nums[i]);
            currMin = min(currMin + nums[i], nums[i]);

            sum += nums[i];

            globalMax = max(globalMax, currMax);
            globalMin = min(globalMin, currMin);
        }

        if (globalMax > 0){
            return max(globalMax, sum-globalMin);
        }
        return globalMax;
    }
};

File: 954-maximum-sum-circular-subarray/maximum-sum-circular-subarray.go
func maxSubarraySumCircular(nums []int) int {
    loc_max, loc_min := 0, 0
    glo_max, glo_min := math.MinInt, math.MaxInt
    sum := 0
    for i := range nums {
        loc_max = max(loc_max + nums[i], nums[i])
        loc_min = min(loc_min + nums[i], nums[i])

        sum += nums[i]

        glo_max = max(glo_max, loc_max)
        glo_min = min(glo_min, loc_min)
    }

    if glo_max > 0 {
        return max(glo_max, sum - glo_min)
    }
    return glo_max
}